core/src/main/java/org/mule/runtime/core/internal/context/DefaultMuleContext.java

```diff
@@ -307,6 +307,10 @@ public void initialise() throws InitialisationException {
 
         lifecycleStrategy.initialise(this);
 
+        // if there is a correlation id generator, check validity. If there is an error in the generation
+        // we want to catch it in deploy time, and avoid waiting for the first event to appear in runtime
+        getConfiguration().getDefaultCorrelationIdGenerator().ifPresent(generator -> generator.validateGenerator());
```

Comment:
add a TODO to move this validation to the validations, when the expressionManager is available there.